src/wp-admin/includes/class-wp-debug-data.php

```diff
@@ -241,92 +241,104 @@ public static function debug_data() {
 			$wp_environment_type = __( 'Undefined' );
 		}
 
+		// Check WP_RUNTIME_ENVIRONMENT.
+		if ( defined( 'WP_RUNTIME_ENVIRONMENT' ) ) {
```

Comment:
Should we be checking `wp_get_runtime_environment()` here as it may be an environmental variable and not a constant?

Reply:
@Clorith thoughts?

Reply:
Are we differentiating between a constant and an environmental variable in these instances?

Reply:
Yes, I think we should check `wp_get_runtime_environment()`.  I think when I added that I just saw what was currently being done for `WP_ENVIRONMENT_TYPE` and adapted it.  But as you suggest, I think Site Health should also call `wp_get_environment_type()` as well.

I'd suggest something like:

```
$wp_runtime_environent = wp_get_runtime_environment();
if ( ! $wp_runtime_environement ) {
	$wp_runtime_environment = __( 'Undefined' );
}
```

Reply:
These are checked/tested under `Info > WordPress`. Making the appropriate updates there.

Reply:
This section is specifically for constant values though (values defined in `wp-config.php` or equivalent), so I believe you shouldn't care what the output of certain functions may be, that should fall to other areas if that's the case, the value in this section is seeing explicitly what is declared, especially as constants can not be re-declared.

If a value may be used to determine the output of a function call that needs to be explicitly output somehow, that should then be added elsewhere.

Using the example of the environment type here, you'll notice the constant section outputs the constant value, that's what matters, while the value of the function is output in the "WordPress" section instead, because it's not a constant defined in the code, but a value interpreted, and used, by WordPress :)

Reply:
Is it worth adding something like this to `default-constants.php`?

```php
if ( ! defined( 'WP_RUNTIME_ENVIRONMENT' ) && function_exists( 'getenv' ) ) {
    $runtime_env = getenv( 'WP_RUNTIME_ENVIRONMENT' );

    if ( $runtime_env ) {
        define ( 'WP_RUNTIME_ENVIRONMENT', $runtime_env );
    }
}
```

If the environment variable is set, they become synonymous, otherwise the constant stays undefined.

Reply:
As I found out after more digging and as @Clorith said, this should be in the `WordPress` section, not under `WP Constants`.

Reply:
Should be fixed in https://github.com/WordPress/wordpress-develop/pull/2225/commits/4f67620eb40bf0e705bbb66f9b882ae8d229738c